### PR TITLE
Update Temurin jck trigger to use PIPELINE_DISPLAY_NAME and optimise test groups submission

### PR DESCRIPTION
**Pull request by andrew-m-leonard**
Identify the AQA_TEST_PIPELINE submitted with \<tag\>:\<platform\>:\<targets\>

Submit non-parallel targets all together to reduce target number of jobs.
Add APPLICATION_OPTIONS param with Temurin exclude list

### CODE REVIEW

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 1]
```diff
@@ -480,31 +480,81 @@ class Build {
...
+                    def displayName = "${buildConfig.SCM_REF} : ${platform} : ${targetsParallel}"
+                    def targets = ""
+                    targetsParallel.each{ target ->
+                        if (targets == "") {
+                            targets = "${target}"
+                        } else {
+                            targets = "${targets},${target}"
+                        }
+                    }
+                    context.triggerRemoteJob abortTriggeredJob: true,
```
[the remark this answers is not in the document]
Comment (andrew-m-leonard) [reply pipelines/build/common/openjdk_build_pipeline.groovy:509]: ah ok that's clever thanks :-)

**Comment by sophia-guo** [comment]
Feels like there is duplicate block. To reduce the duplication could we use the map so no need to check the parallel target size and duplicate the code, something like:

```
def targets = [ 'serial': 'sanity.jck, extended.jck, special.jck']
if ("${platform}" == 'x86-64_linux' || "${platform}" == 'x86-64_windows' || "${platform}" == 'x86-64_mac') {
    targets['serial'] =   'sanity.jck, special.jck'
    targets.put('parallel' : 'extended.jck')
 }

targets.each {
    target -> 
             try {
                context.println "Remote trigger ${target.value}"
                remoteTargets["${target.value}"] = {
                    def displayName = "${buildConfig.SCM_REF} : ${platform} : ${target.value}"
                    def parallel = 'None'
                    def num_machines = '1'
                    if ( target.key == 'parallel') {
                         parallel = 'Dynamic'
                         num_machines = '2'
                    }

                    context.triggerRemoteJob abortTriggeredJob: true,
                        blockBuildUntilComplete: false,
                        job: 'AQA_Test_Pipeline',
                        parameters: context.MapParameters(parameters: [context.MapParameter(name: 'SDK_RESOURCE', value: 'customized'),
                                                                context.MapParameter(name: 'TARGETS', value: ${target.value}),
                                                                context.MapParameter(name: 'CUSTOMIZED_SDK_URL', value: "${sdkUrl}"),
                                                                context.MapParameter(name: 'JDK_VERSIONS', value: "${jdkVersion}"),
                                                                context.MapParameter(name: 'PARALLEL', value: parallel),
                                                                context.MapParameter(name: 'NUM_MACHINES', value: "${num_machines}"),
                                                                context.MapParameter(name: 'PLATFORMS', value: "${platform}"),
                                                                context.MapParameter(name: 'PIPELINE_DISPLAY_NAME', value: "${displayName}"),
                                                                context.MapParameter(name: 'LABEL_ADDITION', value: additionalTestLabel)]),
                        remoteJenkinsName: 'temurin-compliance',
                        shouldNotFailBuild: true,
                        token: 'RemoteTrigger',
                        useCrumbCache: true,
                        useJobInfoCache: true
                }
            } catch (Exception e) {
                context.println "Failed to remote trigger jck tests: ${e.message}"
            }
}

```

**Comment by andrew-m-leonard** [comment]
@sophia-guo yeah, having looked at it I think your single block looks better, i'll re-work it, thanks

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 2]
```diff
@@ -480,31 +480,40 @@ class Build {
...
-                        parallel = 'Dynamic'
-                        num_machines = '2'
+                context.println "Remote trigger: ${targetValue}"
+                remoteTargets["${targetValue}"] = {
+                    def displayName = "${buildConfig.SCM_REF} : ${platform} : ${targetValue}"
+                    def parallel = 'None'
+                    def num_machines = '1'
+                    if ("${targetKey}" == 'parallel') {
+                         parallel = 'Dynamic'
+                         num_machines = '2'
```
Comment (karianna) [inline pipelines/build/common/openjdk_build_pipeline.groovy:503]: Can we go higher?  Should it be a passed in value?

Comment (andrew-m-leonard) [reply pipelines/build/common/openjdk_build_pipeline.groovy:503]: @karianna  I think the parallel or not is an aspect of the remote test triggering, the caller is simply requesting the "platform" to trigger on...
unless you meant something different?

**Comment by andrew-m-leonard** [comment]
@sophia-guo i've removed the duplication. I'm going to also add the customJtx=<exclusion list> option setting to this PR as well.

**Comment by andrew-m-leonard** [comment]
Depends on https://github.com/adoptium/aqa-tests/pull/4196
for APPLICATION_OPTIONS support

**Comment by andrew-m-leonard** [comment]
I have pointed the temurin-compliance jenkins AQA_TEST_PIPELINE at my branch for https://github.com/adoptium/aqa-tests/pull/4196, so we can at least run this this weekend.

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 3]
```diff
@@ -480,31 +481,60 @@ class Build {
...
+        if (platform.contains("aix")) {
+            excludePlat = "aix"
+        } else if (platform.contains("mac")) {
+            excludePlat = "mac"
+            excludeRoot = "/Users"
+        } else if (platform.contains("windows")) {
+            excludePlat = "windows"
+            excludeRoot = "c:/Users"
+        } else if (platform.contains("solaris")) {
+            excludePlat = "solaris"
```
Comment (zdtsw) [inline pipelines/build/common/openjdk_build_pipeline.groovy:499]: is solairs using "/home" or "/export/home"  as `excludeRoot`

Comment (andrew-m-leonard) [reply pipelines/build/common/openjdk_build_pipeline.groovy:499]: good point, i'll check!

Comment (andrew-m-leonard) [reply pipelines/build/common/openjdk_build_pipeline.groovy:499]: yep changed, thanks

**Comment by andrew-m-leonard** [comment]
run tests

**Review by zdtsw — APPROVED** [review]

File: pipelines/build/common/openjdk_build_pipeline.groovy (hunk 2 above)
Comment (sxa) [reply pipelines/build/common/openjdk_build_pipeline.groovy:524]: @karianna 2 seems to be optimal for meeting our targets with the systems we have. Happy to discuss this offline if you wish.

**Review by sxa — COMMENTED** [review]
Will this still allow application options to be overridden when running a job? Will those values to appended to the existing ones or can it replace them if I didn't want the exclusions?

**Review by sophia-guo — APPROVED** [review]
LGTM

**Pull request merged**